dateparser/utils.py

```diff
@@ -86,3 +86,24 @@ def find_date_separator(format):
     m = re.search(r'(?:(?:%[dbBmaA])(\W))+', format)
     if m:
         return m.group(1)
+
+
+class Registry(type):
```

Comment:
@Allactaga I tried to achieve the same using factory pattern and decorators but this looks like a lot cleaner solution. Can you please suggest the alternate implementation?

On a side note, why do you think subclassing it with one's own metaclass will be tempting?